crates/red_knot_python_semantic/src/semantic_index/use_def.rs

```diff
@@ -348,10 +340,12 @@ impl<'db> UseDefMap<'db> {
     fn declarations_iterator<'a>(
         &'a self,
         declarations: &'a SymbolDeclarations,
+        may_be_undeclared: bool,
     ) -> DeclarationsIterator<'a, 'db> {
         DeclarationsIterator {
             all_definitions: &self.all_definitions,
             inner: declarations.iter(),
+            may_be_undeclared,
```

Comment:
Can't this be
```suggestion
            may_be_undeclared: declarations.maybe_undeclared,
```

You may need to flip the order of `inner` and `may_be_undeclared` for the borrow checker.

Reply:
Good call! Nah, borrow checker has no issues here, this was just a result of rewriting the code too many times :) The original version didn't have `may_be_undeclared` stored on the `DeclarationsIterator` at all.